pygfunction/gfunction.py

```diff
@@ -1681,16 +1691,28 @@ def thermal_response_factors(self, time, alpha, kind='linear'):
             # -----------------------------------------------------------------
             if i+1 < nBoreholes:
                 # Segments of the emitting borehole
-                b1 = [seg
-                      for b in self.boreholes[i+1:]
-                      for seg in b.segments(self.nSegments)]
+                b1 = []
+                for b_i in range(i+1, nBoreholes):
+                    nq_b_i = self.nSegments[b_i]
+                    b1.extend(self.boreholes[b_i].segments(nq_b_i))
                 h = finite_line_source(time, alpha, b1, b2)
                 # Broadcast values to h_ij matrix
                 for j in range(i+1, nBoreholes):
-                    j0 = j*self.nSegments
-                    j1 = j0 + self.nSegments
+                    if type(self.nSegments) == int:
+                        nq_j = self.nSegments
+                    else:
+                        nq_j = self.nSegments[j]
+                    if type(self.nSegments) == int:
+                        j0 = j * nq_j
+                        j1 = j0 + nq_j
+                    else:
+                        # sum all previous segments up to borehole i
+                        j0 = sum(self.nSegments[0:j])
+                        # add the number of segments in borehole i
+                        j1 = j0 + nq_j
                     h_ij[i0:i1, j0:j1, 1:] = h[:, j0-i1:j1-i1, :]
                     if j > i:
+                        # TODO: Resolve segment length scaling for reciprocal response
                         H_ratio = self.boreholes[i].H/self.boreholes[j].H
                         h_ij[j0:j1, i0:i1, 1:] = np.transpose(
                             h[:, j0-i1:j1-i1, :]*H_ratio, (1, 0, 2))
```

Comment:
I have nearly added in the ability to compute g-functions when `nSegments` is off and the `method` is detailed. However, one thing I'm not quite understanding is how the reciprocity relationship could have ever been scaled by H[i]/H[j], when that is the ratio of borehole height to borehole height, _not_ a vector of segment heights to segment heights.

Reply:
Wouldn't this work (?) : `H_ratio = (self.boreholes[i].H / nq_i) / (self.boreholes[j].H / nq_j)`

Reply:
Agreed. Your idea works for this branch, I referred to it as a short-term resolution 4357392eb5ed93a07675c5c23a5e89cd603d0989 

I plan to move to #137 now. Let me know if you'd rather have #137 merge into #133 or #135. If not I'll open a PR for #137 after I make it happen.

Reply:
I don't see how #137 is a separate issue from either of #133 and #135 -- it is only an issue in the context of these issues. Changes to the reciprocal responses are within the scope of #133 and #135. I would simply close #137. We can still make PRs to #133 and #135.

Reply:
I believe I have properly addressed this with b61c2879de6be2411a523ec70c9689a9050b57e5. The solution maintains speed via broadcasting. There is going to be slightly more memory consumed because I load `segment_lengths` into memory, however in #135 I plan to include a new instance variable named `segmentLengths` that stores the data long term.